### DISCUSSION TITLE
config-secrets-exposed error on commitFilesToBranch

### DISCUSSION THREAD

**Discussion by ocristian** [renovatebot/renovate#8202] [Request Help] [answered]
Hi there. 
I started to test renovate, self-hosted, to update dependencies on bitbucket-server. For this test, I'm running renovate on docker and using env vars to set the platform, user/pass, endpoint, GitHub token... 

The first run was ok, I got the onboard PR. After merging the onboard PR, I ran renovate again, now to scan my dependencies and create new PR with the updates, but I'm having the error, config -secrets-exposed, and it's failing to commit files to the branch.

The error:

```
DEBUG: 1 file(s) to commit (repository=dep/demo, branch=**redacted**/com.fasterxml.jackson.core-jackson-core-2.x)
ERROR: Error updating branch: config-secrets-exposed (repository=dep/demo, branch=**redacted**/com.fasterxml.jackson.core-jackson-core-2.x)
       "err": {
         "message": "config-secrets-exposed",
         "stack": "Error: config-secrets-exposed\n    at Object.commitFilesToBranch (/usr/src/app/node_modules/**redacted**/lib/workers/branch/commit.ts:43:11)\n    at Object.processBranch (/usr/src/app/node_modules/**redacted**/lib/workers/branch/index.ts:469:29)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at Object.writeUpdates (/usr/src/app/node_modules/**redacted**/lib/workers/repository/process/write.ts:49:17)\n    at Object.update (/usr/src/app/node_modules/**redacted**/lib/workers/repository/process/extract-update.ts:128:11)\n    at Object.**redacted**Repository (/usr/src/app/node_modules/**redacted**/lib/workers/repository/index.ts:45:17)\n    at Object.start (/usr/src/app/node_modules/**redacted**/lib/workers/global/index.ts:67:7)\n    at /usr/src/app/node_modules/**redacted**/lib/**redacted**.ts:10:22"
       }
```

I've checked the file `/lib/workers/branch/commit.ts:43:11` but I couldn't find the root cause. It may be related to `secrets` but I'm not sure as I'm passing them as env var.
Here is where the problem happens... 
```
  if (
    config.branchName !== sanitize(config.branchName) ||
    config.commitMessage !== sanitize(config.commitMessage)
  ) {
    throw new Error(CONFIG_SECRETS_EXPOSED);
  }
```

My config.json
```
{
  "logLevel" : "trace",
  "gitAuthor": "Renovate Bot Test <renovate-bot@test.com>" 
}
```

How I'm running renovate
```
docker run --rm \
  -v "/path_to/renovate/config.json:/usr/src/app/config.json" \
  -e RENOVATE_PLATFORM=bitbucket-server \
  -e RENOVATE_ENDPOINT=http://bitbucket:7990 \
  -e RENOVATE_GIT_FS=http \
  -e RENOVATE_USERNAME=renovate-bot@test.com \
  -e RENOVATE_PASSWORD=****** \
  -e RENOVATE_AUTODISCOVER=true \
  -e GITHUB_COM_TOKEN=******************** \
  -e LOG_LEVEL=trace \
  --net net-renovate \
  -l bitbucket:bitbucket \
  renovate/renovate
  ```

Any ideas? I checked the documents, but I couldn't find the solution ... I may be missing some configuration... not sure.
Let me know if you need any other info.
Thanks 👍🏻

**Comment by viceice** [comment, marked as answer]
I guess the password for your renovate bot user is equal to the renovate branch prefix. That why renovate is complaining about exposed secrets. You should really use a more complex password

**Reply by ocristian** [reply]
Hi @viceice 
Thanks for the quick answer.

You're completely right, that was the problem. As it was a local test I kept the credentials very simple... I was wondering why it didn't happen with the onboard PR as the branch name had the same prefix: `renovate/configure`...

Anyway, it's working now.
Thanks for your time. 👍🏻
